src/gridfs-stream/upload.ts

```diff
@@ -118,7 +144,7 @@ export class GridFSBucketWriteStream extends Writable {
    * @returns {Promise<void>} if no callback specified
    */
 
-  abort(callback: Callback) {
+  abort(callback: Callback<undefined>): Promise<void> | void {
```

Comment:
This should really use `maybePromise`

```suggestion
abort(): Promise<DeleteResult>;
  abort(callback: Callback<DeleteResult | undefined>): void;
  abort(callback?: Callback<DeleteResult | undefined>): Promise<DeleteResult> | void {
    return maybePromise(callback, callback => {
      if (this.state.streamEnd) {
        const error = new Error('Cannot abort a stream that has already completed');
        return callback(error);
      }
      if (this.state.aborted) {
        const error = new Error('Cannot call abort() on a stream twice');
        return callback(error);
      }
      this.state.aborted = true;
      return this.chunks.deleteMany({ files_id: this.id }, callback);
    });
  }
```